### ISSUE TITLE
Implement clipboard support on mobile

### ISSUE THREAD

**Issue by ddkwork** [cogentcore/core#710]
![Screenshot_20231226_163543_com.org.todo.files.jpg](https://github.com/goki/gi/assets/19886504/f5b32317-c47e-4f50-abe7-f6b3de4aad05)

On Android devices, the text edit box does not have a double-click to select all text and a right-click to pop up the copy to clipboard menu

**Comment by kkoreilly** [comment]
Yes, I have been planning to implement copying on mobile and will do so soon. Thanks for reporting this.

**Comment by kkoreilly** [comment]
Why are you closing issues?

**Comment by ddkwork** [comment]
> Why are you closing issues?

The reports here (https://github.com/cogentcore/core/discussions/858) go on for a long time and are panic levels rather than time-consuming operations like widget expansion. This report bothered me for a long time, just like the previous Vulkan Panic, and we both understood what was causing it, and we knew how to write the code to fix it, and why we didn't fix it. In scenarios such as ERP, database, packet capture, etc., as long as these situations are triggered by back-end business-intensive tasks, can you fix any panic reports first? Instead of ignoring it outright. I don't like the verbosity of rust and the generic syntax of go, but it reduces the number of panic occurrences, don't you think?

**Comment by ddkwork** [comment]
Last night I added a lot of lock code to this demo, and finally came to the place where I saw the map code before, which is really memorable and memorable. After changing the 4 null pointers, 3 interface assertion checks, and 2map lock protection, no matter how the lock is in the last local variable where the map is locked, the map read and write exception is triggered, causing the demo to exit, and the interface is stuck and cannot be clicked.

**Comment by kkoreilly** [comment]
I apologize for all of the panics and the misunderstandings. I will focus on fixing all of the panics now. If you want to file a PR with your changes you said fixed certain panics, that would be helpful. Otherwise, I will work on debugging and fixing all of the panics you have mentioned.

**Comment by ddkwork** [comment]
I'm trying to change the code, but this change may break the existing data structure and compatibility, and you need to verify it further, and when the demo I said is no longer panic and runs smoothly, I believe that some of the issues in all issues will disappear with it



---Original---
From: ***@***.***&gt;
Date: Fri, Feb 16, 2024 10:14 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [cogentcore/core] Implement clipboard support on mobile (Issue#710)




 
I apologize for all of the panics and the misunderstandings. I will focus on fixing all of the panics now. If you want to file a PR with your changes you said fixed certain panics, that would be helpful. Otherwise, I will work on debugging and fixing all of the panics you have mentioned.

**Comment by kkoreilly** [comment]
I will try any PR and merge it if it works. The nil pointer dereferences and interface assertions should be relatively easy to fix, so I recommend that you do those in the places that it panics first. Please do not add nil checks to everything, only the places where it panics.
